docs/unit-testing.md

```diff
@@ -0,0 +1,60 @@
+# Unit Testing
+
+We utilize [Jest](https://jestjs.io/docs/getting-started) for the purposes of writing unit tests. Unit tests are meant to be small, fast, and should focus on testing isolated pieces of code. Typically these tests will only import a single source file and test exported functions where given a certain input, a certain output is generated. This documentation will go into details on how the team expects these tests to be wrote, best practices, and general DOs and DONTs.
+
+## Running Unit Tests
+
+After making sure you have the project dependencies installed, executing `npm test` will run all the unit tests. To run just a specific test, you can pass the path of the test file to the test command like so `npm test -- src/features/jserrors/aggregate/format-stack-trace.test.js`. There is also a watch command setup `npm run test:watch` if you are making multiple edits to tests and would like those tests to run after each saved change. To check the state of coverage, use `npm run test:coverage`.
+
+## Writing Tests - Basics
+
+If you are create a new test file for an existing or new source file, remember these best practices:
+
+- **DO** name the test file the same as the target source file but with the `.test.js` extension
+- **DO** create the test file in the same directory as the source file
+
+Once you have a test file created, follow these best practices for creating the individual test cases:
+
+- **DONT** create a global `describe` wrapper to hold all the tests
+- **DO** create a `describe` to group tests that have similar pre/post test instructions
+- **DONT** save state between individual test cases
+- **DO** write and assume that each test case will be ran in isolation, parallelized, and share no local or global state
+- **DO** use `beforeEach` to instantiate variables and state before each test
+- **DO** use `afterEach` to cleanup global state after each test
+- **DONT** use `beforeAll` or `afterAll`
+- **DO** use the `test` keyword to declare a test case
```

Comment:
Curious where these best practices came from

Reply:
These are best practices that we as a team should agree to. They came from my personal experience with writing jest tests.

Reply:
These seem pretty reasonable. Couple questions:
- Why not create a global describe wrapper? If a file happens to contain only one describe wrapper, that's sort of the same thing as a global one in my mind. Seems like the idea is to use describe wrappers only for tests that share setup/tear-down or a clear descriptive relation, and if in doubt split the tests out.
- Why not use beforeAll and afterAll? Because of state spanning test cases?